src/DrDotnet.Profilers/profilers/gc_survivors_profiler.rs

```diff
@@ -327,13 +325,20 @@ impl CorProfilerCallback for GCSurvivorsProfiler
         // Create dependency tree, but from object to referencers, instead of object to its references.
         // This is usefull for being able to browse from any object back to its roots.
         for object_ref_id in object_ref_ids {
+
+            if !Self::is_gen_2(self.clr(), object_ref_id.clone()) {
+                continue;
+            }
+            
             self.object_to_referencers.entry(*object_ref_id)
                 .and_modify(|referencers| referencers.push(object_id))
                 .or_insert(vec![object_id]);
         }
 
         // Also add this object, with no referencers, just in case this object isn't referenced 
-        self.object_to_referencers.insert(object_id, vec![]);
+        if Self::is_gen_2(self.clr(), object_id.clone()) {
```

Comment:
I think we need those, because what we want is to display retention paths for gen 2 objects, which may include non gen 2 objects. For this reason, I don't think we can filter out non gen 2 objects here.

Reply:
When you say "we can't filter out non gen 2 objects here", do you mean at the highlighted line only or rather in the whole `object_references` callback?

I agree with you when you say that we want "to display retention path for gen 2 objects, which could include non gen 2 objects" and because of that I think I should delete the gen 2 test in the for loop also, right?

Reply:
I meant in the whole `object_references`

> I agree with you when you say that we want "to display retention path for gen 2 objects, which could include non gen 2 objects" and because of that I think I should delete the gen 2 test in the for loop also, right?

Unfortunately, yes

I think there is a way to make this an order of magnitude more performant but it might complexify everything even more, so maybe for now we should stick we the latest optims we did (which are quite nice already)